### ISSUE TITLE
[Feature request] Weigh recently modified files higher

### ISSUE THREAD

**Issue by joech4n** [scambier/obsidian-omnisearch#172]
**Is your feature request related to a problem? Please describe.**
I have a harder time finding recently edited files.

**Describe the solution you'd like**
It would be nice to weigh recently modified files higher. Or have a way to specify a specific weight to the file's recency.

**Describe alternatives you've considered**
Using the built-in search.

**Additional context**
N/A

**Comment by tralston** [comment]
I'm following both #123 and this issue, and would add my vote to the others. The majority of work I do in Obsidian is for Work/Meeting notes, which are all date-relevant. Here's a couple of thoughts that might help in updating the algorithm/UI:

- Sorting by recency could either be a search token (e.g. @date, #date, /date, etc.)
- Or it could be through a shortcut key. At the bottom of the Omnisearch window, you have several keyboard shortcuts to toggle between various "modes" of the search. Another one could be recency
- For the algorithm, it would suffice to just sort directly by creation/modification date. So you're either in "Recent" mode or "Recommended" mode.

I'd like to see an option in the settings page where I could set a default mode (Recent or Recommended) and then specify for Recent mode which timestamp to use (Creation / Modification). I can't see people swapping between the two timestamps that often, so a setting seems appropriate. But once people have decided which mode to use as their default for Omnisearch, one of those keyboard shortcuts can toggle between the modes on the fly. And maybe some sort of appearance change or indicator would clue people into which mode they are in at the moment.

**Comment by scambier** [comment]
Ok so basically a "date mode" that completely bypasses the weighting system?

**Comment by tralston** [comment]
> Ok so basically a "date mode" that completely bypasses the weighting system?

Yes. Essentially there are two mindsets I'm in when I'm searching for notes.

1.  When was the most recent thing with a keyword (often in the title, but not always)?
2. What are all the notes I have on a specific topic over all time? (this is the default way you have it querying currently)

**Comment by S-Wil** [comment]
+1 I really need this. Lack of date prioritization is killing me.  I'd also add sometimes I'm not looking for most recently modified but by date created.

**Comment by ovgolovin** [comment]
@scambier Typically, recency is incorporated into search result rankings by using a decay function. Here's an example of how it can be applied in Minisearch, provided by the library's author: https://github.com/lucaong/minisearch/issues/223#issuecomment-1628669680

**Comment by ovgolovin** [comment]
I have just looked, in Elastic search they do decay scoring as a second step after getting the results from first scorring.
They describe it [here](https://www.elastic.co/blog/practical-bm25-part-3-considerations-for-picking-b-and-k1-in-elasticsearch#:~:text=Adding%20or%20using%20a%20function%20score%20to%20decay%20the%20scores%20of%20older%20documents%20or%20documents%20which%20are%20further%20geographically%20from%20the%20end%20user).

**Comment by scambier** [comment]
@ovgolovin thanks for the references, that looks easy enough to implement and toy with 👍

**Comment by scambier** [comment]
A first draft has been published in [1.26.0](https://github.com/scambier/obsidian-omnisearch/releases/tag/1.26.0)

Let's continue discussion at https://github.com/scambier/obsidian-omnisearch/discussions/401
